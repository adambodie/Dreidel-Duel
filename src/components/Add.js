import React, { Component } from "react";
import styles from "./main.module.css";

export default class Add extends Component {
	constructor(props) {
		super(props);
		this.handleChange = this.handleChange.bind(this);
	}

	handleChange(e) {
		this.props.onAddChange(e.target.value);  
	}

	render() {
		return (
			<button className={styles.action} onClick={this.handleChange} disabled={this.props.disabled}>
				Add Coin
			</button>
		);
  }
}
