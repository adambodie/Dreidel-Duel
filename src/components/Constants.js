export const GIMEL = "ג";
export const SHIN = "שׁ";
export const NUN = "נ";
export const HAY = "ה";
