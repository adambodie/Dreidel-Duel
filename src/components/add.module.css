.addAction {
	background: #0000FF;
	margin-bottom: 20px;
	font-weight: bold;
	color: #FAFAFA;
	font-size: 2em;
	padding: 5px 5px;
	border-radius: 15px;
}
