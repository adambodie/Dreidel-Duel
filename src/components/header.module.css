@import url('https://fonts.googleapis.com/css?family=Ravi+Prakash');

.jumbotron {
	padding: 2rem 1rem;
    margin-bottom: 2rem;
    background:url(../dreidel.jpg);
    background-size: cover;
    background-position: bottom;
}
.display4 {
	font-family: 'Ravi Prakash', cursive;
	text-shadow: 5px 5px 10px #000;
	text-align: center;
    font-size: 6rem;
    font-weight: 300;
    line-height: 1.2;
}
.display3 {
	font-family: 'Ravi Prakash', cursive;
	text-shadow: 5px 5px 10px #000;
	text-align: right;
	margin: 0;
}
@media (min-width: 576px) {
	.jumbotron {
		padding: 3rem 1.5rem;
	}
}
