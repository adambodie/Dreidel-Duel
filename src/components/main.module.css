@import url('https://fonts.googleapis.com/css?family=Vollkorn');

body {
	font-family: 'Vollkorn', serif;
	color: #FAFAFA;
	background-color: blue;
}
h2 {
	font-size: 1.5em;
	letter-spacing: 3px;
	font-weight: normal;
	text-align: center;
}

/* BUTTON STYLES
----------------------*/
.buttons {
	display: flex;
	align-items: center;   
}
button {
	display: block;
	margin-left: auto;
	margin-right: auto;
	
	box-shadow: 0 10px #999;
}

button:active {
	box-shadow: 0 5px #666;
	transform: translateY(4px);
}

button:disabled {
	background: #D3D3D3;
}

.action {
	background: #0000FF;
	margin-bottom: 20px;
	font-weight: bold;
	color: #FAFAFA;
	font-size: 2em;
	padding: 5px 5px;
	border-radius: 15px;
}

.action:disabled:hover {
	background-color: #D3D3D3;
}

.action:hover {
	background-color: navy;
}


.players {
	display: flex;
	align-items: center;
	justify-content: space-between; 
}

/* MISCELLANEOUS CSS
 ---------------------*/

.scoreboard {
	margin-left: auto;
	margin-right: auto;
	display: block;
	border: 20px groove yellow;
	border-radius: 10px;
	background:url(../scoreboard.jpg);
	background-size: cover;
	box-shadow: 0 12px 24px 0 rgba(0, 0, 0, 0.8), 0 12px 60px 0 rgba(0, 0, 0, 0.38);
	margin-bottom: 50px;
}

.jumbotron {
	text-align: center;
	background: #0000FF;
	padding: 0;
}

.footer {
	padding: 5px 10px;
	text-align: center;
}
.footer h2 {
	font-size: 3em;
}
span {
	font-size: 8em;
	position: absolute;
	margin-top: -200px;
	margin-left: -50px;
	width: 105px;
}

@media (min-width: 768px) {
	.scoreboard {
		width: 600px;
	}
}
