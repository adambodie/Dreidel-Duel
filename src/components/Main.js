// Libs
import React, { Component } from "react";
import styles from "../css/main.module.css";
import Pot from './Pot';
import Spin from './Spin';
import Reset from './Reset';
import Start from './Start';
import Add from './Add';
import Player from './Player';


export default class Main extends Component {
	constructor() {
		super();
		this.state = {
			pot: 10,
			spin: "",
			turn: 0,
			isDisabled: true,
			isButtonDisabled: true,
			players: [
				{ score: 5, isDisabled: true, id: 1 },
				{ score: 5, isDisabled: true, id: 2 }
			]
		};
		this.onStartChange = this.onStartChange.bind(this);
		this.onResetChange = this.onResetChange.bind(this);
		this.onScoreChange = this.onScoreChange.bind(this);
		this.onAddChange = this.onAddChange.bind(this);
	}

	onStartChange() {
		const newPlayers = [...this.state.players];
		newPlayers[0].isDisabled = false;
		this.setState(prevState => ({
			newPlayers,
			isDisabled: !prevState.isDisabled
		}));
	}
	
	onResetChange() {
		const newPlayers = [...this.state.players];
		newPlayers.forEach((item) => {
			item.isDisabled = true;
			item.score = 5;
		});
		this.setState(prevState => ({
			newPlayers,
			isDisabled: !prevState.isDisabled,
			pot: 10,
			spin: "",
			isButtonDisabled: true,
			turn: 0
		}));
	}
  
	onAddChange() {
		const newPlayers = [...this.state.players];
		let playerOne = newPlayers[0];
		let playerTwo = newPlayers[1];
		if (playerOne.score === 0 || playerTwo.score === 0) {
			newPlayers.forEach((element) => element.isDisabled = true );
			this.setState(prevState => ({
				newPlayers,
				pot: "Game Over, play again"
			}));
		} else {
			if (playerOne.isDisabled === true &&  this.state.turn >= 1) {
				newPlayers.forEach((element) => element.score -= 1 );
				playerOne.isDisabled = false;
				this.setState(prevState => ({ pot: this.state.pot + 2 }));
			}
		}
		this.setState(prevState => ({ isButtonDisabled: true }));
	}
  
	onScoreChange() {
		let dreidel = Math.floor(Math.random() * 4);
		const hebrew = ["נ", "שׁ", "ה", "ג"];
		let currentSpin = hebrew[dreidel];
		const newPlayers = [...this.state.players];
		let playerOne = newPlayers[0];
		let playerTwo = newPlayers[1];

		if (playerOne.isDisabled === true && this.state.turn >= 1) {
			this.setState(prevState => ({ isButtonDisabled: false }));
		}
		if ((playerOne.score <= 0 || playerTwo.score <= 0) && (dreidel === 1 || dreidel === 3)) {
			newPlayers.forEach((element) => element.isDisabled = true );
			this.setState(prevState => ({
				newPlayers,
				pot: "Game Over, play again"
			}));
		} else {
			if (dreidel === 0) {
			/* Spin a Nun - Nothing */
				if (playerOne.isDisabled === false) {
					newPlayers.forEach((element) => element.isDisabled = !element.isDisabled );
				} else {
					playerTwo.isDisabled = true;
				}
			} else if (dreidel === 1) {
			/* Spin a Shin - Put one in the Pot */
				if (playerOne.isDisabled === false) {
					newPlayers.forEach((element) => element.isDisabled = !element.isDisabled );
					playerOne.score -= 1;
				} else {
					playerTwo.score -= 1;
					playerTwo.isDisabled = true;
				}
				this.setState(prevState => ({ pot: this.state.pot + 1 }));
			} else if (dreidel === 2) {
			/* Spin a Hay - Win half of your coins in the Pot */
				if (playerOne.isDisabled === false) {
					newPlayers.forEach((element) => element.isDisabled = !element.isDisabled );
					playerOne.score += Math.round(this.state.pot / 2);
				} else {
					playerTwo.score += Math.round(this.state.pot / 2);
					playerTwo.isDisabled = true;
				}
				this.setState(prevState => ({ pot: Math.floor(this.state.pot / 2) }));
			} else {
			/* Spin a Gimel - Win everything  */
				if (playerOne.isDisabled === false) {
					playerOne.score += this.state.pot;
					newPlayers.forEach((element) => element.isDisabled = !element.isDisabled );
				} else {
					playerTwo.score += this.state.pot;
					playerTwo.isDisabled = true;
				}
				newPlayers.forEach((element) => element.score  -= 1 );
				this.setState(prevState => ({ pot: 2 }));
			}
			this.setState(prevState => ({
				newPlayers,
				spin: currentSpin,
				turn: this.state.turn + 1
			}));
		}
	}

  render() {
    return (
		<div className={styles.container}>
			<div className={styles.players}>
				<Player 
					score={this.state.players[0].score} 
					disabled={this.state.players[0].isDisabled} 
					onScoreChange={this.onScoreChange} 
					players={this.state.players}
					id={this.state.players[0].id}
					/>
					<Spin spin={this.state.spin} />
				<Player  
					score={this.state.players[1].score} 
					disabled={this.state.players[1].isDisabled} 
					onScoreChange={this.onScoreChange} 
					players={this.state.players}
					id={this.state.players[1].id}
					/>
			</div>
			<div className={styles.footer}>
				<Add onAddChange={this.onAddChange} disabled={this.state.isButtonDisabled}/>
				<Pot pot={this.state.pot} />
			</div>
			<div className={styles.buttons}>
				<Reset onResetChange={this.onResetChange} disabled={this.state.isDisabled} />
				<Start onStartChange={this.onStartChange} disabled={!this.state.isDisabled} />
			</div>
        </div>
    );
  }
}
