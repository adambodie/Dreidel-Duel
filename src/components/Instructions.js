import React, { Component } from 'react';
import { NUN, SHIN, HAY, GIMEL } from './Constants';
import Modal from 'react-modal';
import ReactTooltip from 'react-tooltip';
import FontAwesome from 'react-fontawesome';
import styles from "../css/header.module.scss";

const customStyles = {
	content : {
		top                   : '50%',
		left                  : '50%',
		right                 : 'auto',
		bottom                : 'auto',
		transform             : 'translate(-50%, -50%)'
	}
};

Modal.setAppElement('#___gatsby')

export default class Instructions extends Component {
	constructor() {
		super();
		this.state = {
			modalIsOpen: false
		};

		this.openModal = this.openModal.bind(this);
		this.closeModal = this.closeModal.bind(this);
  }

	openModal() {
		this.setState({modalIsOpen: true});
	}

	closeModal() {
		this.setState({modalIsOpen: false});
	}

	render() {
		return (
			<div>
				<FontAwesome 
					onClick={this.openModal}
					name='question-circle'
					className={styles.question}
					size='3x'
				/>
				<Modal
					isOpen={this.state.modalIsOpen}
					onRequestClose={this.closeModal}
					style={customStyles}
					contentLabel="Instructions"
				>
				<div className={styles.modal}>
					<header className={styles.modalHeader}>
						<h2>Instructions</h2>
						<FontAwesome 
							onClick={this.closeModal} 
							name='window-close'
							className={styles.close}
							size='2x'
						/>
					</header>
					<ReactTooltip />
					<div className={styles.instructions}>
						<p>Spin the dreidel.  What happens depends on which side it lands.</p>
						<p>When the last player of each turn plays, each player places a coin in the pot.</p>
						<p>If there are no coins in the pot, each player places a coin in the pot.</p>
						<p>When a player has no coins and is unable to gain any coins, they are eliminated from the game.</p>
						<div className={styles.instructionFlex}>
							<p><span data-tip="Gimel">{GIMEL}</span> - Take all the coins in the pot.</p>
							<p><span data-tip="Hay">{HAY}</span> - Take half of the coins in the pot.</p>
							<p><span data-tip="Shin">{SHIN}</span> - Put one coin in the pot.</p>
							<p><span data-tip="Nun">{NUN}</span> - Do nothing.</p>
						</div>
					</div>
				</div>
			</Modal>
		</div>
		);
	}
}
