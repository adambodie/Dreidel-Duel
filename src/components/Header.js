import React from "react"
import styles from "../css/header.module.scss";
import Instructions from "./Instructions";
import { StaticQuery, graphql } from "gatsby";

export const PureHeader = ({ data }) => (
		<div className={styles.jumbotron}>
			<h1 className={styles.display4}>{data.site.siteMetadata.title}</h1>
			<h3 className={styles.display3}>by {data.site.siteMetadata.author}</h3>
			<Instructions />
		</div>
)


export const Header = props => (
  <StaticQuery
    query={graphql`
      query {
        site {
          siteMetadata {
            title
            author
          }
        }
      }
    `
    }
	render={data => <Header {...props} data={data} />}
/>
)

export default Header
