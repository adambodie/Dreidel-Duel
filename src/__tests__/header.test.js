import React from "react"
import renderer from "react-test-renderer"
import { PureHeader as Header } from "../components/Header"
import {shallow} from "enzyme"

describe("Header", () =>
	it("renders correctly", () => {
		const data = {
			site: {
				siteMetadata: {
					title: "Dreidel Duel!",
					author: "Adam Bodie"
				},
			},
		}	  
		const tree = renderer.create(<Header data={data}/>).toJSON()
		expect(tree).toMatchSnapshot()
	}),
	it('renders a `button`', () => {
		const data = {
			site: {
				siteMetadata: {
					title: "Dreidel Duel!",
					author: "Adam Bodie"
				},
			},
		}
		const wrapper = shallow(<Header data={data} />);
		expect(wrapper.find('.jumbotron').length).toBe(1);
	}),
	it('renders a `button`', () => {
		const data = {
			site: {
				siteMetadata: {
					title: "Dreidel Duel!",
					author: "Adam Bodie"
				},
			},
		}
		const wrapper = shallow(<Header data={data} />);
		expect(wrapper.find('.jumbotron').children().length).toBe(2);
	}),	  
   
)
