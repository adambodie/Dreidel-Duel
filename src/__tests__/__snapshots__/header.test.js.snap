// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header renders correctly 1`] = `
<div
  className="jumbotron"
>
  <h1
    className="display4"
  >
    Dreidel Duel!
  </h1>
  <h3
    className="display3"
  >
    by 
    Adam Bodie
  </h3>
</div>
`;
